### DISCUSSION TITLE
Zero vertical emittance in presence of lattice errors

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#727] [Python AT]
Dear AT Expert,

I encountered an issue in the value of emittance, during tuning simulation i performed for the FCCee lattice. Specifically, I assigned alignment errors to some quadrupoles and sextupoles and followed a correction scheme while the radiation is off. after that, I calculated the emittance value of the lattice after correction using ohmi_envelope() function .

However, the vertical emittance value was consistently zero for all 20 seeds,  This happend when introducing 10 and 20 micrometer alignment errors, and it only began to show a slight increase with larger error values.

This consistent zero value for  in the presence of lattice errors has raised concerns on our end. We are uncertain whether this issue stems from AT or if there might be a misperformance on our part.

I have attached to this message one lattice seed and the script for optics calculation. 
[fccee_emittance_value_with_err.zip](https://github.com/atcollab/at/files/13981449/fccee_emittance_value_with_err.zip)

Thank you in advance!

**Comment by simoneliuzzo** [comment]
Dear @elafmusa,

I will look at it now. I agree with you. It seems a classic missing `ring.atenable_6d()`.

ciao
simone

**Comment by simoneliuzzo** [comment]
Dear @elafmusa,

I do not know why ohmi_envelope gives zero vertical emittance

```
params = ring.envelope_parameters()

emittance_h = params.emittances[0]
emittance_v = params.emittances[1]
sigma_delta =  params.sigma_e
print(f'eh {emittance_h*1e12:.2f} pm')
print(f'ev {emittance_v*1e12:.2f} pm')

```
gives the vertical emittance

@lfarv may be know why this is the case?

best regards
Simone

**Comment by lcarver** [comment]
Hello, 

Following on from Simones comment. In 'envelope_parameters' it is directly calling 'ohmi_envelope' like this: 

```
emit0, beamdata, emit = ohmi_envelope(ring, orbit=orbit, keep_lattice=keep_lattice)
rp.emittances = beamdata.mode_emittances

```

However in your script you call it like this:

```
emit0, bbb, eee = ring.ohmi_envelope()
emittance_h = emit0['emitXY'][0]
emittance_v = emit0['emitXY'][1]
```

which is the wrong attribute.

Instead use bbb.mode_emittances in your script for the correct emittance.

**Comment by elafmusa** [comment]
Dear @simoneliuzzo and @lcarver ,
Thank you for the clarification! i have modified my script to obtain the correct emittance values.
